src/NServiceBus.Core/Pipeline/BehaviorContext.cs

```diff
@@ -8,16 +9,21 @@
     /// </summary>
     public abstract class BehaviorContext : ContextBag
     {
+        //TODO DanielTim: Does it make to force us to provide a RootContext here (to ensure Settings are set)? RootContext currently is private.
         /// <summary>
         /// Create an instance of <see cref="BehaviorContext"/>.
         /// </summary>
         /// <param name="parentContext">The parent context.</param>
-        protected BehaviorContext(ContextBag parentContext) : base(parentContext)
+        protected BehaviorContext(BehaviorContext parentContext) : base(parentContext)
         {
-          
         }
 
         /// <summary>
+        /// The readonly settings.
+        /// </summary>
+        public ReadOnlySettings Settings => Get<ReadOnlySettings>();
```

Comment:
The biggest issue is the `PipelineBase` constructor. We just remove the DI magic for `IMessageMapper` and floated it into the settings so that we can access it over the settings